modules/c/examples/code_snippets/main.c

```diff
@@ -1018,7 +1100,44 @@ static void enable_basic_auth() {
     CBLAuth_Free(basicAuth);
 
     CBLReplicator_Start(replicator, false);
-    // end::basic-authentication[]
+    // end::basic-authentication-full[]
+}
+
+static void docsonly_N1QL_Params() {
+    CBLDatabase* database = kDatabase;
+
+    // tag::query-syntax-n1ql-params[]
+    int errorPos;
+
```

Comment:
Why are there spaces between lines?  Is this to facilitate the overlaid numbers on the rendered snippets?  Otherwise it just wastes space and looks odd.

Reply:
Habit probably. For a doco whitespace is a good thing. It isolates and highlights and can make things easier to scan read.
As regards code I also tend to group things in 'paragraphs'; grouping related commands. So maybe setting up the args for a function call and the call itself would be one para ... I'll cut it out as much as possible but there will be a difference between 'example' code and how a pro would write production code ... in addition to my Spanglish approach to coding :)

Reply:
There is a lot of room for style in this, just as you would write an essay for English class.  However, putting so many newlines in like this is the equivalent of writing one sentence paragraphs, which most English teachers would frown on.  I don't think I've ever seen production code written like this, and I was worried that there was some sort of tool sporadically inserting the whitespace when I saw it.  Grouping things together into sort of logical action steps is the normal way I separate things.